dspace-api/src/main/java/org/dspace/util/FunctionalUtils.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.util;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca at 4science.com)
+ *
+ */
+public class FunctionalUtils {
```

Comment:
Please add JavaDocs to describe.  For instance, how are `FunctionalUtils` different from other utility classes under `org.dspace.utils.*` or even `org.dspace.core.Utils`

Reply:
As described with the `JavaDocs`:
> These methods are linked to the functional paradigm and use {@code Functional} interfaces of java 8+, all the main
> interfaces are in the package {@link java.util.function}.